#!/usr/bin/env Rscript
#This script uses as input the files generated by download-SRA-convert-paired.fastq.R (fastq paired files and sampleFile.txt) and then aligns the data
library("QuasR")

qA <- qAlign("sampleFile.txt", "BSgenome.Hsapiens.UCSC.hg19") #create the alignment based on BSgenome.Hsapiens.UCSC.hg19
save(qA, file = "qAlign.RData") #save the qProject variable to outfile qAlign.RData
#load("qAlign.RData") #in case session is destroyed

#qC <- qCount(qA, NULL, reportLevel="junction") #mode 4/4 test, exon-exon junctions
#write.table(qC, "qCount.csv", sep=",") #write result to qCount.csv
#save(qC, file = "qAlign.RData") #save the qC variable if write.table fails

