# RNA-seq
.sra data handling with R in Linux

# download-SRA-convert-paired-fastq.R

* Please install the fastq-dump for linux beforehand and add fastq-dump to the PATH.

Downloads experiment files (.sra).
Needs SraRunInfo.csv to be downloaded before execution.
From https://www.ncbi.nlm.nih.gov/sra query your dataset and then: Send to (top right corner), Select “File”,  Select format “RunInfo”,  Click on “Create File".
This script also creates the sampleFile.txt which is needed as input for the next script (alignment and counting).

# alignment.R
* Please install the QuasR package beforehand.

#This script uses as input the files generated by download-SRA-convert-paired.fastq.R (fastq paired files and sampleFile.txt) and then aligns the data
#Produces BAM files

# transcriptome_assembly.R
* Please install cufflinks for linux beforehand and add cufflinks to the PATH.

#This script uses as input the BAM files generated by the qAlign alignment and then assembles the transcriptome via cufflinks for each file in the respective folder
#Must be executed in the same directory where the .bam files were generated

# differential_expression.R
#This script uses as input the files generated by cufflinks and returns the transcript and gene counts for each experiment
#cuffcompare and cuffdiff must be installed in linux (they come with cufflinks) and set in the PATH

#UNDER CONSTRUCTION
