#!/usr/bin/env Rscript
#This script uses as input the BAM files generated by the qAlign alignment and then assembles the transcriptome via cufflinks for each file in the respective folder
#Must be executed in the same directory where the .bam files were generated
#cufflinks must be installed in linux and set in the PATH

files <- list.files(pattern = "\\.bam$")
for(f in files) { #for all generated bam files
	system2("cufflinks", files) #run cufflinks for final output
	exp_name <- substr(f,0,10)
	folder_name <- paste(exp_name,"_folder",sep="") #new folder to be created for each experiment
	dir.create(file.path(getwd(), folder_name) #moving to corresponding folder by name
	target_dir_file <- paste(exp_name,"transcripts.gtf",sep="/") #moving all cufflinks generated files one by one
	file.rename(from="transcripts.gtf", to=target_dir_file)
	target_dir_file <- paste(exp_name,"isoforms.fpkm_tracking",sep="/")
	file.rename(from="isoforms.fpkm_tracking.gtf", to=target_dir_file)
	target_dir_file <- paste(exp_name,"genes.fpkm_tracking",sep="/")
	file.rename(from="genes.fpkm_tracking", to=target_dir_file)
	target_dir_file <- paste(exp_name,"skipped.gtf",sep="/")
	file.rename(from="skipped.gtf", to=target_dir_file)
}
